Backpack/src/main/java/net/skyscanner/backpack/calendar/view/CalendarView.kt

```diff
@@ -94,8 +95,10 @@ internal class CalendarView constructor(
     previousScrollState = currentScrollState
 
     listener?.onYearChanged(child.getYear())
+    scrollListener?.onScroll(this, firstVisibleItem, visibleItemCount, totalItemCount)
```

Comment:
I think this would work better if we remove the current `OnYearChangedListener` and allow more than one `scrollListener` to be added. This new scroll listener should provide the current year as the last argument as well.

```kotlin
interface BpkCalendarScrollListener {
   fun onScroll(view: AbsListView, firstVisibleItem: Int, visibleItemCount: Int, totalItemCount: Int, year: Int)
}
```

```kotlin
fun addScrollListener(listener: BpkCalendarSscrollListener) {
  listeners.add(listener)
}
fun removeScrollListener(listener: BpkCalendarSscrollListener) {
   listeners.remove(listener)
}
```

And in that case remove the custom  `OnYearChangedListener`